### DISCUSSION TITLE
Is it possible to use Xamarin Forms WebView in a Uno project?

### DISCUSSION THREAD

**Discussion by vadd98** [unoplatform/uno#4971] [Q&A] [answered]
I'm currently developing a cross-platform (Windows and Mac) app using Uno and WinUI 3. I need to implement a WebView in my project, but I'm having some trouble. On Windows I can use the new WebView2 from WinUI without any problem, but being it not supported by Uno I can't use it on Mac. On Mac I think I can't even use the WebView1 because it is not supported anymore in WinUI 3 (and so in Uno.WinUI, or am I wrong?). So I thought to use the Xamarin Forms WebView on Mac, but I don't know how to implement it. I thought to add a new project to the solution with just the WebView and call it from the Uno.Mac project, but it doesn't work. May you give me any suggestions, please?

**Comment by jeromelaban** [comment]
Thanks for the report.

Could you provide a simple repro sample that demonstrates what's not working ? You can use a default `dotnet new unoapp` template, or a Visual Studio 2019 Uno Platform template.

**Comment by vadd98** [comment]
I'll try to explain better with some code references.
I'm writing a desktop application for Windows and Mac using WinUI 3 and Uno.WinUI. I need to implement a WebView and on Windows I implemented the WebView2 from WinUI without any problem. On Mac I can not use the WebView2 because it is not supported by Uno.WinUI, so I'm trying to use the Xamarin.Forms WebView as a replacement exploiting the Xamarin.Mac engine used for the Mac Uno project. I created a Class Library with just the Xamarin.Forms WebView in it named "WebView_Xamarin_Mac".
I'm trying to start the WebView with the press of a button. Here's an extract of the code from the Mac project:

```
using Microsoft.UI.Xaml;
using Microsoft.UI.Xaml.Controls;
using System.Threading.Tasks;
using System;
using Xamarin.Forms;
using AppKit;
using WebView_Xamarin_Mac;

namespace Project
{
    public sealed partial class Home : Microsoft.UI.Xaml.Controls.Page
    {
        public Home()
        {
            this.InitializeComponent();
        }

        private void ButtonPressed(object sender, RoutedEventArgs e)
        {
            Xamarin.Forms.Forms.Init(); // initialize Xamarin.forms

            NSApplication.SharedApplication.Delegate = new WebView_Xamarin_Mac.AppDelegate(); // call the AppDelegate from the Mac Class Library

            Xamarin.Forms.NavigationPage navigationPage = new NavigationPage(new WebView_Xamarin_Mac.WebWindow()); // navigate to the WebView
        }
}
```

This code is actually executed, but nothing really happens or shows up. By the way I do not know if this approach is correct or not. Any suggestions?

**Comment by jeromelaban** [comment, marked as answer]
I see. Indeed WebView2 is not supported at all at this point, and the Xamarin.Forms Uno Renderers are not yet compatible with Uno.WinUI.

I created https://github.com/unoplatform/uno/issues/4774 for the tracking of WinUI 3.0 support for Xamarin.Forms, but currently, the easiest way for you to achieve this goal is to use Uno.UI, not Uno.WinUI.

**Comment by stefanolson** [comment]
Is WebView2 still not supported?  I can't get anything to appear using WebView2, where as WebView works fine.
